src/UsgsAstroLsSensorModel.cpp

```diff
@@ -750,14 +749,15 @@ csm::ImageCoord UsgsAstroLsSensorModel::groundToImage(
 
    double preSquare = desired_precision * desired_precision;
    if (warnings && (desired_precision > 0.0) && (preSquare < len)) {
-      std::stringstream msg;
-      msg << "Desired precision not achieved. ";
-      msg << len << "  " << preSquare << "\n";
-      warnings->push_back(
-         csm::Warning(csm::Warning::PRECISION_NOT_MET,
-         msg.str().c_str(),
-         "UsgsAstroLsSensorModel::groundToImage()"));
+     MESSAGE_LOG(m_logger, "groundToImage: Desired precision not achieve {}", preSquare)

```

Comment:
missing the d on achieved

Reply:
Also, this should be logged even when the warning list is not passed by the caller. I'd move the creation of the working to a nested if block. Also, you can re-use the logging string in the warning.